src/client/pmix_client.c

```diff
@@ -552,21 +552,10 @@ PMIX_EXPORT pmix_status_t PMIx_Abort(int flag, const char msg[],
         /* shouldn't be possible */
         goto done;
     }
-#if defined(PMIX_ENABLE_DSTORE) && (PMIX_ENABLE_DSTORE == 1)
```

Comment:
@igor-ivanov @rhc54 @jjhursey this should fix the problem with functionality degradation when dstore is enabled,

The logic is the following:
- we always put locally submitted data to the hash
- at pmix_get we check hash if the key belongs to **thi_rank** or the rank is undefined.